library/ix-dev/community/qbittorrent/questions.yaml

```diff
@@ -0,0 +1,209 @@
+groups:
+  - name: qBittorrent Configuration
+    description: Configure qBittorrent
+  - name: User and Group Configuration
+    description: Configure User and Group for qBittorrent
+  - name: Network Configuration
+    description: Configure Network for qBittorrent
+  - name: Storage Configuration
+    description: Configure Storage for qBittorrent
+  - name: Resources Configuration
+    description: Configure Resources for qBittorrent
+
+portals:
+  web_portal:
+    protocols:
+      - "$kubernetes-resource_configmap_portal_protocol"
+    host:
+      - "$kubernetes-resource_configmap_portal_host"
+    ports:
+      - "$kubernetes-resource_configmap_portal_port"
+    path: "$kubernetes-resource_configmap_portal_path"
+
+questions:
+  - variable: qbitConfig
+    label: ""
+    group: qBittorrent Configuration
+    schema:
+      type: dict
+      attrs:
+        - variable: additionalEnvs
+          label: Additional Environment Variables
+          description: Configure additional environment variables for qBittorrent.
+          schema:
+            type: list
+            default: []
+            items:
+              - variable: env
+                label: Environment Variable
+                schema:
+                  type: dict
+                  attrs:
+                    - variable: name
+                      label: Name
+                      schema:
+                        type: string
+                        required: true
+                    - variable: value
+                      label: Value
+                      schema:
+                        type: string
+                        required: true
+
+  - variable: qbitRunAs
+    label: ""
+    group: User and Group Configuration
+    schema:
+      type: dict
+      attrs:
+        - variable: user
```

Comment:
This should not be allowed to change as well after the app has been deployed?

Reply:
Technically can change. As long as the volumes are chownd again. 

I can see use case for it: 
A separate app that needs to access the same data, can't have its user changed. But you can then match qbit's user. 


Maybe we just add that info in the description/readme?

Reply:
To confirm we would be changing ownership of the volumes again if run context changes?

Reply:
Currently that's not the case. We only chown at startup. 
Meaning that if user/group is changed, user will have to do the chown manually

Reply:
@stavros-k don't you think it should be either of the following:
1. Do not allow user to change run context
2. Change ownerships if run context changes

We could maybe just check if parent directory has different owner and then do chown on updates if we go with (2)?

Reply:
If we go with (2), we might run into issues like parent dir having correct owner (in case user have manually changed it, or another app did) but sub-dir/files does not.
Resulting in a kinda broken state, which user in the end will have to chown manually. 

If we go with (1), can we later (if we need to) remove the `immutable` flag on upgrade?

Reply:
@stavros-k we can remove that flag yes and i think (2) is not a huge problem either as the mismatch will only happen when the run context is changed and not otherwise so checking parent dir will be enough..

Reply:
Added a new init container for addressing (2). Maybe we should change this as the default behavior on all init containers for permissions?